app/models/foreman_azure_rm/azure_rm.rb

```diff
@@ -212,6 +222,12 @@ def vm_nics(vm)
       ifaces
     end
 
+    def vm_disks(vm)
+      vols = []
+      vm.storage_profile.data_disks.map { |disk| vols << disk  }
```

Comment:
I am missing something here. `data_disks` is already an array, either we are missing some transformation or we don't need the `map` at all and we can return `data_disks` directly

Reply:
I agree, we can directly return `data_disks` here.